wasm-chain-integration/src/lib.rs

```diff
@@ -271,7 +271,15 @@ impl State {
     }
 }
 
-pub struct InitHost<'a> {
+pub struct InitHost<'a, Ctx, P, A>
+where
+    A: AsRef<[u8]>,
+    P: SerialPolicies<A>,
+    Ctx: HasInitContext<P>, {
```

Comment:
I would recommend removing these constraints from the struct definition. This would also remove the P and A types, and the phantom fields.